src/metricscollector/collector/apppoller.go

```diff
@@ -0,0 +1,83 @@
+package collector
+
+import (
+	"metricscollector/cf"
+	"metricscollector/db"
+	"metricscollector/metrics"
+
+	"code.cloudfoundry.org/lager"
+	"time"
+)
+
+type AppPoller struct {
+	logger   lager.Logger
+	appId    string
+	interval time.Duration
+	cfc      cf.CfClient
+	noaa     NoaaConsumer
+	database db.DB
+	ticker   *time.Ticker
+	doneChan chan bool
+}
+
+func NewAppPoller(logger lager.Logger, appId string, interval time.Duration, cfc cf.CfClient, noaa NoaaConsumer, database db.DB) *AppPoller {
+	return &AppPoller{
+		logger:   logger,
+		appId:    appId,
+		interval: interval,
+		cfc:      cfc,
+		noaa:     noaa,
+		database: database,
+		doneChan: make(chan bool),
+	}
+
+}
+
+func (ap *AppPoller) Start() {
+	ap.ticker = time.NewTicker(ap.interval)
+	go ap.startPollMetrics()
+
+	ap.logger.Info("app-poller-started", lager.Data{"appid": ap.appId})
+}
+
+func (ap *AppPoller) Stop() {
+	if ap.ticker != nil {
+		ap.ticker.Stop()
+		ap.doneChan <- true
```

Comment:
Normally, you would close the channel to signify its done rather than send a value.
The other issue is that you made it a non-buffered channel which means the go routine will wait here until something pulls it off. While it will work, it does cause a delay.
Either switch to close(app.doneChan) or make the channel buffered with 1 element.